ContactMechanics/Systems.py

```diff
@@ -518,3 +519,133 @@ def minimize_proxy(self, solver=constrained_conjugate_gradients, **kwargs):
 
             self.substrate.check()
         return result
+
+    def primal_objective(self, offset, pot=False, gradient=True):
+        r"""To solve the primal objective using gap as the variable.
+        Can be fed directly to standard solvers ex: scipy solvers etc
+        and returns the elastic energy and it's gradient (negative of
+        the forces) as a function of the gap.
+
+        Parameters
+        __________
+
+        gap : gap between the contact surfaces.
+        offset : "constant value to add to the surface heights
+        pot : (default False)
+        gradient : (default True)
+
+        Returns
+        _______
+        energy : value of energy(scalar value).
```

Comment:
Please adhere to numpydoc for documentation strings. This would be
```
energy : float
    value of energy (scalar value)
```

Reply:
Apologies. I have updated the doc strings now as per numpydoc.

Reply:
Thanks! And no need to apologize, the review process requires a certain pickiness...